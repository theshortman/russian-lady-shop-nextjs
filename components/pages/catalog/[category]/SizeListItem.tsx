import { FC } from "react";

interface SizeListItemProps {
  size: number;
}

const SizeListItem: FC<SizeListItemProps> = ({ size }) => {
  return <span className="badge bg-secondary me-1">{size}</span>;
};

export default SizeListItem;
