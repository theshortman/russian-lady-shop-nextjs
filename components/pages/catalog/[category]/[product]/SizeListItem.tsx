import { FC } from "react";

interface SizeListItemProps {
  size: number;
  quantity: number;
}

const SizeListItem: FC<SizeListItemProps> = ({ size, quantity }) => {
  return (
    <label className="me-2">
      <input className="me-1" type="radio" name="size" value={quantity} />
      {size}
    </label>
  );
};

export default SizeListItem;
