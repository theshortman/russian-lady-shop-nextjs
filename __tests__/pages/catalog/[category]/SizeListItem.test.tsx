import { render } from "@testing-library/react";
import SizeListItem from "@/pages/catalog/[category]/SizeListItem";

describe("Category page", () => {
  test("renders SizeListItem unchanged", () => {
    const size = { size: 50, quantity: 1 };

    const { container } = render(<SizeListItem {...size} />);
    expect(container).toMatchSnapshot();
  });
});
