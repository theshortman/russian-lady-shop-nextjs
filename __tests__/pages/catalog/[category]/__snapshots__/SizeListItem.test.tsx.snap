// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category page renders SizeListItem unchanged 1`] = `
<div>
  <span
    class="badge bg-secondary me-1"
  >
    50
  </span>
</div>
`;
